pkg/cosign/verify.go

```diff
@@ -620,20 +620,72 @@ func verifySignatures(ctx context.Context, sigs oci.Signatures, h v1.Hash, co *C
 func verifyInternal(ctx context.Context, sig oci.Signature, h v1.Hash,
 	verifyFn signatureVerificationFn, co *CheckOpts) (
 	bundleVerified bool, err error) {
+	var acceptableRFC3161Time, acceptableRekorBundleTime *time.Time // Timestamps for the signature we accept, or nil if not applicable.
+
+	if co.TSACerts != nil {
+		acceptableRFC3161Timestamp, err := VerifyRFC3161Timestamp(ctx, sig, co.TSACerts)
+		if err != nil {
+			return false, fmt.Errorf("unable to verify RFC3161 timestamp bundle: %w", err)
+		}
+		if acceptableRFC3161Timestamp != nil {
+			acceptableRFC3161Time = &acceptableRFC3161Timestamp.Time
+		}
+	}
+
+	if !co.SkipTlogVerify {
+		bundleVerified, err = VerifyBundle(ctx, sig, co, co.RekorClient)
+		if err != nil {
+			return false, fmt.Errorf("error verifying bundle: %w", err)
+		}
+
+		if bundleVerified {
+			// Update with the verified bundle's integrated time.
+			t, err := getBundleIntegratedTime(sig)
+			if err != nil {
+				return false, fmt.Errorf("error getting bundle integrated time: %w", err)
+			}
+			acceptableRekorBundleTime = &t
+		} else {
+			// If the --offline flag was specified, fail here. bundleVerified returns false with
+			// no error when there was no bundle provided.
+			// TODO: You can be offline when you are verifying with a public key. This shouldn't
```

Comment:
@haydentherapper already mentioned this to me, this can be removed at this point (use the --skip flag if you want to skip)

Reply:
Removed